R/target_market_share.R

```diff
@@ -220,6 +221,60 @@ target_market_share <- function(data,
     ) %>%
     select(-.data$target_name, -.data$green_or_brown)
 
+  summary_groups <- c(
+    "scenario",
+    "region",
+    "scenario_source",
+    "name_ald"
+  )
+
+  if (weight_production) {
+    data <- data %>%
+      ungroup() %>%
+      add_loan_weight(use_credit_limit = use_credit_limit) %>%
+      add_technology_share() %>%
+      add_technology_share_target() %>%
+      calculate_weighted_loan_metric("production") %>%
+      calculate_weighted_loan_metric("technology_share") %>%
+      calculate_weighted_loan_metric("production_target") %>%
+      calculate_weighted_loan_metric("technology_share_target") %>%
+      group_by(
+        .data$sector_ald,
+        .data$technology,
+        .data$year,
+        !!!rlang::syms(summary_groups)
+      ) %>%
+      summarize(
+        weighted_production = sum(.data$weighted_loan_production),
+        weighted_technology_share = sum(.data$weighted_loan_technology_share),
+        weighted_production_target = sum(.data$weighted_loan_production_target),
+        weighted_technology_share_target = sum(.data$weighted_loan_technology_share_target)
```

Comment:
It's sometimes useful to call `sum()` with `rm.na = TRUE`. Likely you took this into account but just checking you didn't forget.